app/services/directions.js

```diff
@@ -0,0 +1,24 @@
+// Google gets loaded in index.html, so is always on the global scope here.
+export const { LatLng, TravelMode } = google.maps;
+const { DirectionsStatus } = google.maps;
+const DirectionsService = new google.maps.DirectionsService();
+
+const determineResult = (resolve, reject) => (result, status) => {
+  if(status === DirectionsStatus.OK) {
+    resolve(result.routes[0].legs[0].duration.value);
+  } else {
+    reject(`Google returned error ${status}`);
+  }
+}
```

Comment:
hmm this is kinda weird. Took me a mo to figure this out. I think I would call this `promisifyResult` or something?